### PR TITLE
Create POST /todos api 

### PR DESCRIPTION
**Pull request by faysal-alam-sazim**
### Description of Change
- Created a task dto file
- Added Post /todos api to insert data to db
### Associated Ticket Link:
- https://trello.com/c/u6KIoHaC

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -1,8 +1,9 @@
 import { Module } from '@nestjs/common';
 import { MikroOrmModule } from '@mikro-orm/nestjs';
+import { TasksModule } from './tasks/tasks.module';
 import mikroOrmConfig from '../mikro-orm.config';
 
 @Module({
-  imports: [MikroOrmModule.forRoot(mikroOrmConfig)],
+  imports: [MikroOrmModule.forRoot(mikroOrmConfig), TasksModule],
```
Comment (ishmam-iqbal-sazim) [inline src/app.module.ts:7]: May need to rebase properly. Shouldn't be seeing changes from other branches

File: src/tasks/tasks.controller.ts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import { CreateTaskDto } from './tasks.dto';
+
+@Controller('todos')
+export class TasksController {
+  constructor(private readonly tasksService: TasksService) {}
+
+  @Get()
+  async findAll(): Promise<Task[]> {
+    return this.tasksService.findAll();
+  }
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.controller.ts:19]: Same comment as above

File: src/tasks/tasks.dto.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  @IsNotEmpty()
+  @IsString()
+  priority: string;
+
+  @IsNotEmpty()
+  @IsString()
+  status: string;
+}
+
+export class UpdateTaskDto extends PartialType(CreateTaskDto) {}
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.dto.ts:28]: You'd need the id here as well

Comment (ishmam-iqbal-sazim) [reply src/tasks/tasks.dto.ts:28]: Disregard this comment.

File: src/db/migrations/.snapshot-to-do.json
Comment (ragib-hossain-sazim) [inline src/db/migrations/.snapshot-to-do.json:1]: same comment as previous PR migration snapshot can be gitIgnored.

File: src/tasks/tasks.service.ts [hunk 4]
```diff
@@ -1,16 +1,25 @@
...
   ) {}
+
   async findAll(): Promise<Task[]> {
-    return this.taskRepository.findAll();
+    return await this.taskRepository.findAll();
+  }
+
+  async createTask(createTaskDto: CreateTaskDto): Promise<Task> {
+    const task = this.taskRepository.create(createTaskDto);
+    await this.em.persistAndFlush(task);
```
Comment (ragib-hossain-sazim) [inline src/tasks/tasks.service.ts:22]: Please move `em` actions to your `tasksRepository` as mentioned in previous PR. the method name in `tasksRepository` should be `createOne`. Service should only contain business logic, database transaction related code should be inside repository.

File: src/tasks/tasks.service.ts [hunk 5]
```diff
@@ -1,12 +1,18 @@
...
 @Injectable()
 export class TasksService {
   constructor(private readonly taskRepository: TaskRepository) {}
+
   async findAll(): Promise<Task[]> {
-    return this.taskRepository.getAllTasks();
+    return await this.taskRepository.findAll();
+  }
+
+  async createTask(createTaskDto: CreateTaskDto): Promise<Task> {
```
Comment (ishmam-iqbal-sazim) [inline src/tasks/tasks.service.ts:15]:
```suggestion
  async createOne(createTaskDto: CreateTaskDto): Promise<Task> {
```

Don't need task in the method name.

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comment.

File: src/db/migrations/.snapshot-to-do.json
Comment (ragib-hossain-sazim) [inline src/db/migrations/.snapshot-to-do.json:1]: Why snapshots are still in the commit?

**Pull request merged**